Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibsAPI.cs

```diff
@@ -0,0 +1,1006 @@
+            }
+        }
+
+        /// <summary>
+        /// Determines whether the specified VBA class implements a specific interface
+        /// </summary>
+        /// <param name="projectTypeLib">Low-level ITypeLib wrapper representing the VBA project</param>
+        /// <param name="className">Document class name, as declared in the VBA project</param>
+        /// <param name="interfaceProgID">The interface name, preceeded by the library container name, e.g. "Excel._Worksheet"</param>
+        /// <returns>bool indicating whether the class does inherit the specified interface</returns>
+        public static bool DoesClassImplementInterface(TypeLibWrapper projectTypeLib, string className, string interfaceProgID)
+        {
+            return DoesClassImplementInterface(projectTypeLib.TypeInfos.Get(className), interfaceProgID);
+        }
+
+        /// <summary>
+        /// Determines whether the specified VBA class implements a specific interface
+        /// </summary>
+        /// <param name="component">Safe-com wrapper representing the VBA component</param>
+        /// <param name="interfaceProgID">The interface name, preceeded by the library container name, e.g. "Excel._Worksheet"</param>
+        /// <returns>bool indicating whether the class does inherit the specified interface</returns>
+        public static bool DoesClassImplementInterface(IVBComponent component, string interfaceProgID)
+        {
+            return DoesClassImplementInterface(component.ParentProject, component.Name, interfaceProgID);
+        }
+
+        /// <summary>
+        /// Determines whether the specified VBA class implements a specific interface
+        /// </summary>
+        /// <param name="classTypeInfo">Low-level ITypeInfo wrapper representing the VBA project</param>
+        /// <param name="interfaceProgID">The interface name, preceeded by the library container name, e.g. "Excel._Worksheet"</param>
+        /// <returns>bool indicating whether the class does inherit the specified interface</returns>
+        public static bool DoesClassImplementInterface(TypeInfoWrapper classTypeInfo, string interfaceProgID)
+        {
+            return classTypeInfo.ImplementedInterfaces.DoesImplement(interfaceProgID);
+        }
+
+        /// <summary>
+        /// Determines whether the specified VBA class implements a specific interface
+        /// </summary>
+        /// <param name="ide">Safe-com wrapper representing the VBE</param>
+        /// <param name="projectName">VBA Project name, as declared in the VBE</param>
+        /// <param name="className">Document class name, as declared in the VBA project</param>
+        /// <param name="interfaceProgIDs">An array of interface names, preceeded by the library container name, e.g. "Excel._Worksheet"</param>
+        /// <param name="matchedIndex">on return indicates the index into interfaceProgIDs that matched</param>
```

Comment:
It might be good to note explicitly that it will be `-1` when it's not matched (though we should be checking the return value anyway before we try to use that parameter). If updated, note to update the overloads.

Reply:
will do, thanks